src/app/clusters/scenes-server/SceneTableImpl.cpp

```diff
@@ -22,6 +22,97 @@
+    // TLV::TLVType outer;
+
+    reader.Init(serializedBytes);
+    // ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag());
+    // ReturnErrorOnFailure(reader.EnterContainer(outer));
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Array, TLV::AnonymousTag()));
+    ReturnErrorOnFailure(aVlist.Decode(reader));
+    // ReturnErrorOnFailure(reader.ExitContainer(outer));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR
+DefaultSceneHandlerImpl::SerializeAdd(EndpointId endpoint,
+                                      const app::Clusters::Scenes::Structs::ExtensionFieldSet::DecodableType & extensionFieldSet,
+                                      MutableByteSpan & serializedBytes)
+{
+    app::Clusters::Scenes::Structs::AttributeValuePair::Type aVPairs[kMaxAvPair];
+
+    size_t pairTotal = 0;
+    // Verify size of list
+    ReturnErrorOnFailure(extensionFieldSet.attributeValueList.ComputeSize(&pairTotal));
+    VerifyOrReturnError(pairTotal <= ArraySize(aVPairs), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    uint8_t pairCount  = 0;
+    auto pair_iterator = extensionFieldSet.attributeValueList.begin();
+    while (pair_iterator.Next())
+    {
+        app::Clusters::Scenes::Structs::AttributeValuePair::DecodableType aVPair = pair_iterator.GetValue();
+        aVPairs[pairCount].attributeID                                           = aVPair.attributeID;
+        aVPairs[pairCount].attributeValue                                        = aVPair.attributeValue;
+        pairCount++;
+    }
+    ReturnErrorOnFailure(pair_iterator.GetStatus());
+    app::DataModel::List<app::Clusters::Scenes::Structs::AttributeValuePair::Type> attributeValueList(aVPairs);
+    attributeValueList.reduce_size(pairCount);
+
+    return EncodeAttributeValueList(attributeValueList, serializedBytes);
+}
+
+CHIP_ERROR DefaultSceneHandlerImpl::Deserialize(EndpointId endpoint, ClusterId cluster, const ByteSpan & serializedBytes,
+                                                app::Clusters::Scenes::Structs::ExtensionFieldSet::Type & extensionFieldSet)
+{
+    app::DataModel::DecodableList<app::Clusters::Scenes::Structs::AttributeValuePair::DecodableType> attributeValueList;
+
+    ReturnErrorOnFailure(DecodeAttributeValueList(serializedBytes, attributeValueList));
+
+    // Verify size of list
+    size_t pairTotal = 0;
+    ReturnErrorOnFailure(attributeValueList.ComputeSize(&pairTotal));
+    VerifyOrReturnError(pairTotal <= ArraySize(mAVPairs), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    uint8_t pairCount  = 0;
+    auto pair_iterator = attributeValueList.begin();
+    while (pair_iterator.Next())
+    {
+        app::Clusters::Scenes::Structs::AttributeValuePair::DecodableType decodePair = pair_iterator.GetValue();
+        mAVPairs[pairCount].attributeID                                              = decodePair.attributeID;
+        mAVPairs[pairCount].attributeValue                                           = decodePair.attributeValue;
```

Comment:
Again, any reason not to do:
```suggestion
        mAVPairs[pairCount] = pair_iterator.GetValue();
```
?